modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/logsdb/qa/StandardVersusLogsIndexModeChallengeRestIT.java

```diff
@@ -0,0 +1,264 @@
+
+        Matcher.mappings(getContenderMappings(), getBaselineMappings())
+            .settings(getContenderSettings(), getBaselineSettings())
+            .expected(getAggregationBuckets(queryBaseline(searchSourceBuilder), "agg"))
+            .ignoringSort(true)
+            .isEqualTo(getAggregationBuckets(queryContender(searchSourceBuilder), "agg"));
+    }
+
+    public void testTermsAggregation() throws IOException, MatcherException {
+        final List<XContentBuilder> documents = new ArrayList<>();
+        int numberOfDocuments = randomIntBetween(100, 200);
+        for (int i = 0; i < numberOfDocuments; i++) {
+            documents.add(generateDocument(Instant.now().plus(i, ChronoUnit.SECONDS)));
+        }
+
+        final Tuple<Response, Response> tuple = indexDocuments(() -> documents, () -> documents);
+        assertThat(tuple.v1().getStatusLine().getStatusCode(), Matchers.equalTo(RestStatus.OK.getStatus()));
+        assertThat(tuple.v2().getStatusLine().getStatusCode(), Matchers.equalTo(RestStatus.OK.getStatus()));
+
+        final SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder().query(QueryBuilders.matchAllQuery())
+            .size(0)
+            .aggregation(new TermsAggregationBuilder("agg").field("host.name"));
+
+        Matcher.mappings(getContenderMappings(), getBaselineMappings())
+            .settings(getContenderSettings(), getBaselineSettings())
+            .expected(getAggregationBuckets(queryBaseline(searchSourceBuilder), "agg"))
+            .ignoringSort(true)
+            .isEqualTo(getAggregationBuckets(queryContender(searchSourceBuilder), "agg"));
+    }
+
+    public void testDateHistogramAggregation() throws IOException, MatcherException {
+        final List<XContentBuilder> documents = new ArrayList<>();
+        int numberOfDocuments = randomIntBetween(100, 200);
+        for (int i = 0; i < numberOfDocuments; i++) {
+            documents.add(generateDocument(Instant.now().plus(i, ChronoUnit.SECONDS)));
+        }
+
+        final Tuple<Response, Response> tuple = indexDocuments(() -> documents, () -> documents);
+        assertThat(tuple.v1().getStatusLine().getStatusCode(), Matchers.equalTo(RestStatus.OK.getStatus()));
+        assertThat(tuple.v2().getStatusLine().getStatusCode(), Matchers.equalTo(RestStatus.OK.getStatus()));
+
+        final SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder().query(QueryBuilders.matchAllQuery())
+            .aggregation(AggregationBuilders.dateHistogram("agg").field("@timestamp").calendarInterval(DateHistogramInterval.SECOND))
+            .size(0);
+
+        Matcher.mappings(getContenderMappings(), getBaselineMappings())
+            .settings(getContenderSettings(), getBaselineSettings())
+            .expected(getAggregationBuckets(queryBaseline(searchSourceBuilder), "agg"))
+            .ignoringSort(true)
+            .isEqualTo(getAggregationBuckets(queryContender(searchSourceBuilder), "agg"));
+    }
```

Comment:
Maybe also add qa test the GET api? 
And is the plan to later also qa test es|ql?

Reply:
Will add those as follow-ups PRs.